pkg/apps/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -542,50 +574,49 @@ func triggerActivated(config *appsapi.DeploymentConfig, latestExists bool, lates
 				// TODO: Technically this is not a config change cause, but we will have to report the image that caused the trigger.
 				//       In some cases it might be difficult because config can have multiple ICT.
 				appsutil.RecordConfigChangeCause(config)
-				return true, false
+				return true, false, nil
 			}
 			glog.V(4).Infof("Rolling out initial deployment for %s/%s deferred until its images are ready", config.Namespace, config.Name)
-			return false, true
+			return false, true, nil
 		}
 		// Rollout if we only have config change trigger.
 		if configTrigger {
 			glog.V(4).Infof("Rolling out initial deployment for %s/%s", config.Namespace, config.Name)
 			appsutil.RecordConfigChangeCause(config)
-			return true, false
+			return true, false, nil
 		}
 		// We are waiting for the initial RC to be created.
-		return false, false
+		return false, false, nil
 	}
 
 	// Wait for the RC to be created
 	if !latestExists {
-		return false, true
+		return false, false, nil
 	}
 
 	// We need existing deployment at this point to compare its template with current config template.
 	if latestDeployment == nil {
-		return false, false
+		return false, false, nil
 	}
 
 	if imageTrigger {
 		if ok, imageNames := appsutil.HasUpdatedImages(config, latestDeployment); ok {
 			glog.V(4).Infof("Rolling out #%d deployment for %s/%s caused by image changes (%s)", config.Status.LatestVersion+1, config.Namespace, config.Name, strings.Join(imageNames, ","))
 			appsutil.RecordImageChangeCauses(config, imageNames)
-			return true, false
+			return true, false, nil
 		}
 	}
 
 	if configTrigger {
 		isLatest, changes, err := appsutil.HasLatestPodTemplate(config, latestDeployment, codec)
 		if err != nil {
-			glog.Errorf("Error while checking for latest pod template in replication controller: %v", err)
-			return false, true
+			return false, false, fmt.Errorf("error while checking for latest pod template in replication controller: %v", err)
```

Comment:
why returning this error? isn't logging it enough?

Reply:
if you return error it will get rate limited if it's repeating and it should not update status as it failed to act on it

Reply:
appsutil.HasLatestPodTemplate can only fail when decoding deployment config, I guess it won't even get this far if decoding fail...

Maybe we can convert the second bool to an error and then check that error time (ErrSkipTrigger) or something?

Reply:
> appsutil.HasLatestPodTemplate can only fail when decoding deployment config, I guess it won't even get this far if decoding fail...

which is likely a permanent failure which the ratelimiting queue will shield us from

> Maybe we can convert the second bool to an error and then check that error time (ErrSkipTrigger) or something?

seems too obscure; skipping the trigger is not an error

but if it fails and gets here, the queue rate limitting needs to kick in because i